boards/common/arduino-atmega/Makefile.dep

```diff
@@ -3,3 +3,5 @@ USEMODULE += boards_common_atmega
 ifneq (,$(filter saul_default,$(USEMODULE)))
   USEMODULE += saul_gpio
 endif
+
+-include $(RIOTCPU)/$(CPU)/Makefile.dep
```

Comment:
I think this is better placed in `boards/common/atmega/Makefile.dep`, as not all AVR based boards are Arduinos. (And this and every non-Arduino AVR board should then include `$(RIOTBOARD)/common/atmega/Makefile.dep`).

Also, I would prefer to use `include $(RIOTCPU)/$(CPU)/Makefile.dep` instead of `-include` here, so that developers adding support for new ATmega CPUs will get an error if they did not provide a corresponding `Makefile.dep`.